cpp/src/Slice/Scanner.l

```diff
@@ -315,47 +316,59 @@ floating_literal    (({fractional_constant}{exponent_part}?)|((\+|-)?{dec}+{expo
 
   /* ========== Comments ========== */
 
-  /* Matches and records a triple-slash style doc comment. */
+  /* Matches and records a triple-slash style doc-comment. */
 <*>"///".* {
     currentUnit->addToDocComment(yytext + 3);
 }
 
   /* Matches and consumes a C++ style comment. */
 <*>"//".* {}
 
+  /* Matches the start of a C style doc-comment, and switches the scanner to the C_DOC_COMMENT state.
+   * Specifically we match the literal "/**" when it is _not_ followed by either a '/' or '*' character. */
+<*>"/**"/[^/*] {
```

Comment:
This is the flex-regex-style for "match `/**`, then do a positive lookahead (the `/`) for anything that is **not** (the `^`) either `/` or `*`".